jpyinterpreter/src/main/java/org/optaplanner/jpyinterpreter/implementors/FunctionImplementor.java

```diff
@@ -524,6 +650,20 @@ public static void createFunction(FunctionMetadata functionMetadata, StackMetada
         MethodVisitor methodVisitor = functionMetadata.methodVisitor;
         String className = functionMetadata.className;
 
+        if (functionMetadata.pythonCompiledFunction.pythonVersion.isAtLeast(PythonVersion.PYTHON_3_11)) {
+            // Python 3.11 and above removed qualified name, so we need to get it from the code object's class
+            methodVisitor.visitTypeInsn(Opcodes.CHECKCAST, Type.getInternalName(PythonCode.class));
+            methodVisitor.visitInsn(Opcodes.DUP);
+            // TODO: maybe create qualifiedName field in PythonCode?
+            methodVisitor.visitFieldInsn(Opcodes.GETFIELD, Type.getInternalName(PythonCode.class), "functionClass",
+                    Type.getDescriptor(Class.class));
+            // TODO: get qualified name from static field?
```

Comment:
Did you leave these TODOs there intentionally?

Reply:
This is for potential future work/refactoring, where instead of getting the qualified name via reflection on the PythonCode's class field, we read a (new) qualifiedName field.